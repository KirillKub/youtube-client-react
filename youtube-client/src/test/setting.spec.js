import React from 'react'
import { shallow } from 'enzyme'

import Setting from '../../src/searchSetting/setting';


describe('Setting', () => {
  it('if isSetting = false, return null', () => {
    const props = {
      isSetting: false,
      sortByDate: () => {},
      sortByCount: () => {},
      sortByInput: () => {},
    }
    const container = shallow(<Setting  {...props}/>);
    expect(container).toMatchSnapshot();
    expect(container.find('div')).toHaveLength(0)  
  })
  it('if isSetting = true, render component', () => {
    const props = {
      isSetting: true,
      sortByDate: () => {},
      sortByCount: () => {},
      sortByInput: () => {},
    }
    const container = shallow(<Setting  {...props}/>);
    expect(container).toMatchSnapshot();
    expect(container.find('div')).toHaveLength(3)
    expect(container.find('span')).toHaveLength(3)  
  })
})