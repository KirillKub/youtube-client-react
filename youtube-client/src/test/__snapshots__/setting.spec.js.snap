// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting if isSetting = false, return null 1`] = `ShallowWrapper {}`;

exports[`Setting if isSetting = true, render component 1`] = `ShallowWrapper {}`;
